test/nools/task-emitter.spec.js

```diff
@@ -95,6 +95,206 @@ describe('task-emitter', () => {
         });
       });
 
+      it('task priority is not required', () => {
+        // given
+        const contact = personWithReports(aReport());
+        const task = aPersonBasedTask();
+        const config = {
+          c: contact,
+          targets: [],
+          tasks: [ task ],
+        };
+        delete config.tasks[0].priority;
+        
+        // when
+        const { emitted } = runNoolsLib(config);
+
+        // then
+        expect(emitted).to.have.length(2);
+        expect(emitted[0]).to.be.shallowDeepEqual({
+          '_type': 'task',
+          'actions': [{
+            'content': {
+              'contact': {
+                '_id': contact.contact._id,
+                'reported_date': TEST_DATE,
+                'type': 'person'
+              },
+              'source': 'task',
+              'source_id': contact.contact._id
+            },
+            'form': 'example-form',
+            'label': 'Follow up',
+            'type': 'report'
+          }],
+          'contact': {
+            '_id': contact.contact._id,
+            'reported_date': TEST_DATE,
+            'type': 'person'
+          },
+          'date': TEST_DAY,
+          'resolved': false
+        });
+      });
+
+      it('task priority is string set from object', () => {
+        // given
+        const report = aReport();
+        const contact = personWithReports(report);
+        const config = {
+          c: contact,
+          targets: [],
+          tasks: [ aPersonBasedTask() ]
+        };
+        config.tasks[0].priority = {
+          level: 'high',
+          label: [{ locale: 'en', label: 'High Priority' }]
+        };
+        
+        // when
+        const { emitted } = runNoolsLib(config);
+
+        // then
+        expect(emitted).to.have.length(2);
+        expect(emitted[0]).to.deep.includes({
+          priority: 'high',
+          priorityLabel: [ { locale: 'en', label: 'High Priority' } ]
+        });
+      });
+
+      it('task priority is string set from callback function', () => {
+        // given
+        const report = aReport();
+        const contact = personWithReports(report);
+        const config = {
+          c: contact,
+          targets: [],
+          tasks: [ aReportBasedTask() ],
+        };
+        config.tasks[0].priority = (c, r, e, d) => {
```

Comment:
I love that you're adding the assertions within the task function. Unfortunately, this will not guarantee that the task function has run, and your test will still succeed even if this function does not run. 
In order to make this comprehensive, you can do something like: 

```suggestion
        config.tasks[0].priority = sinon.stub().callsFake((c, r, e, d) => {
          expect(c).to.be.deep.equals(contact);
          expect(r).to.be.deep.equals(report);
          expect(e).to.be.deep.equals(config.tasks[0].events[0]);
          expect(d).to.be.deep.equals(TEST_DAY);

          return {
            level: 'high',
            label: [ { locale:'en', label: 'High Priority' } ]
          };
        });
        ....<run nools lib>....
        expect(config.tasks[0].priority.calledOnce).to.be.true;
```

Reply:
An alternative is to not run assertions within the function, and would just do something like: 
```suggestion
config.tasks[0].priority = sinon.stub().returns({ 
  level: 'high',
  label: [ { locale:'en', label: 'High Priority' } ]
});
...<<run nools lib>>
expect(config.tasks[0].priority.calledOnce).to.be.true;
expect(config.tasks[0].priority.args[0]).to.deep.equal([contact, report, config.tasks[0].events[0], TEST_DAY]);
```